src/core.ts

```diff
@@ -110,6 +110,23 @@ instance.interceptors.request.use(
   }
 );
 
+instance.interceptors.request.use((config: AxiosRequestConfig) => {
+  const url = config.url || '';
+  const baseUrl = config.baseURL || '';
+  const cogniteUrl = /^https:\/\/.*\.cognitedata.com/;
+  const isUnknownDomain =
+    // rawGet('https://example.com');
+    (/^http/.test(url) && !cogniteUrl.test(url)) ||
+    // rawGet('/path') with baseUrl !== cognite
+    (!cogniteUrl.test(baseUrl) && !cogniteUrl.test(url));
```

Comment:
Lots of false positives and negatives here:

| URL | `isUnknownDomain` | Expected value | Comment |
|-|-|-|-|
| `http/path` | `true` | False | False positive
| `//example.com/path` | `false` | True | **False negative**
| `https://api.cognitedata.com.my-evil-domain.com/path` | `false` | True | **False negaitve**
| `https://api.cognitedata-com.com/path` | `false` | True | **False negaitve**
| `https://my-evil.domain.com/cognitedata.com/path` | `false` | True | **False negative**
| `ftp://my-evil-domain.com` | `false` | True | False negative<sup>1</sup>

<sub><sup>1</sup>This is less important though since major browsers (as far as I know) don't support non-local, non-HTTP(S) URLs in fetch requests.</sub>

Also, it will give false positives when `baseUrl` is set to a non-CDF domain (e.g. when testing locally with `localhost`).

Reply:
My suggested alternative check (with [`url`](https://www.npmjs.com/package/url) package, for instance):

```js
function isSameOrigin(baseUrl, newUrl) {
  const url = require('url');
  const baseUrlParsed = url.parse(baseUrl);
  const newUrlParsed = url.parse(url.resolve(baseUrl, newUrl));

  // check that protocol and hostname are the same
  return baseUrlParsed.protocol === newUrlParsed.protocol
    && baseUrlParsed.host === newUrlParsed.host;
}

const isUnknownDomain = !isSameOrigin(baseUrl, url);
```